packages/xdg_directories/lib/xdg_directories.dart

```diff
@@ -0,0 +1,163 @@
+  assert(envVar != null);
+  final String value = _getenv(envVar);
+  if (value == null || value.isEmpty) {
+    if (fallback == null) {
+      return null;
+    }
+    return _getDefaultDir(fallback);
+  }
+  return Directory(value);
+}
+
+Directory _getDefaultDir(String suffix) {
+  assert(suffix != null);
+  assert(suffix.isNotEmpty);
+  final String homeDir = _getenv('HOME');
+  if (homeDir == null || homeDir.isEmpty) {
+    throw StateError('The "HOME" environment variable is not set. This package (and POSIX) '
+        'requires that HOME be set.');
+  }
+  return Directory(path.joinAll(<String>[homeDir, suffix]));
+}
+
+/// The base directory relative to which user-specific
+/// non-essential (cached) data should be written. (Corresponds to
+/// `$XDG_CACHE_HOME`).
+///
+/// Throws [StateError] if the HOME environment variable is not set.
+Directory get cacheHome => _directoryFromEnvironment('XDG_CACHE_HOME', '.cache');
+
+/// The list of preference-ordered base directories relative to
+/// which configuration files should be searched. (Corresponds to
+/// `$XDG_CONFIG_DIRS`).
+///
+/// Throws [StateError] if the HOME environment variable is not set.
+List<Directory> get configDirs => _directoryListFromEnvironment('XDG_CONFIG_DIRS', '/etc/xdg');
+
+/// The a single base directory relative to which user-specific
+/// configuration files should be written. (Corresponds to `$XDG_CONFIG_HOME`).
+///
+/// Throws [StateError] if the HOME environment variable is not set.
+Directory get configHome => _directoryFromEnvironment('XDG_CONFIG_HOME', '.config');
+
+/// The list of preference-ordered base directories relative to
+/// which data files should be searched. (Corresponds to `$XDG_DATA_DIRS`).
+///
+/// Throws [StateError] if the HOME environment variable is not set.
+List<Directory> get dataDirs => _directoryListFromEnvironment('XDG_DATA_DIRS', '/usr/local/share:/usr/share');
+
+/// The base directory relative to which user-specific data files should be
+/// written. (Corresponds to `$XDG_DATA_HOME`).
+///
+/// Throws [StateError] if the HOME environment variable is not set.
+Directory get dataHome => _directoryFromEnvironment('XDG_DATA_HOME', '.local/share');
+
+/// The base directory relative to which user-specific runtime
+/// files and other file objects should be placed. (Corresponds to
+/// `$XDG_RUNTIME_DIR`).
+///
+/// Throws [StateError] if the HOME environment variable is not set.
+Directory get runtimeDir => _directoryFromEnvironment('XDG_RUNTIME_DIR', null);
+
+/// Gets the xdg user directory named by `dirName`.
+///
+/// Use [getUserDirectoryNames] to find out the list of available names.
+Directory getUserDirectory(String dirName) {
+  final ProcessResult result = _processManager.runSync(
+    <String>['xdg-user-dir', dirName],
+    includeParentEnvironment: true,
+    stdoutEncoding: Encoding.getByName('utf8'),
+  );
+  final String path = utf8.decode(result.stdout).split('\n')[0];
```

Comment:
`trim()` before split, just in case an implementation adds an extra newline before the actual output?

Reply:
I'm going to keep what I have: I don't want to remove any leading/trailing whitespace that might be part of the path, as unlikely as that is.